miso-web/src/main/webapp/scripts/library_hot.js

```diff
@@ -8,12 +8,16 @@ Library.hot = {
   libraryTypeAliases: {},
   
   getLibraryTypeAliasLists: function () {
-    Hot.dropdownRef.libraryTypes.forEach(function (lt) {
-      if (Library.hot.libraryTypeAliases[lt.platform]) {
-        Library.hot.libraryTypeAliases[lt.platform].push(lt.alias);
-      } else {
-        Library.hot.libraryTypeAliases[lt.platform] = [lt.alias];
+    var usedTypes = Library.hot.librariesJSON.map(function (lib) {
+      return lib.libraryTypeId;
+    });
+    Hot.dropdownRef.libraryTypes.filter(function (lt) {
+      return !lt.archived || usedType.indexOf(lt.id) != -1;
```

Comment:
You'll want to add `archived` to the list of LibraryType attributes which get written in the EditLibraryController's `referenceDataJsonString` method (or refactor the whole method?).